### PR TITLE
Add new api getFungibleAssetMetadata

### PR DESCRIPTION
**Pull request by 0xjinn**
### Description

Add the new query getFungibleAssetMetadata. 

Note: This time I am modifying on how we construct the function in `/internal` Please take a look

### Test Plan

pnpm fmt
pnpm lint
pnpm test

### CODE REVIEW

**Comment by 0xjinn** [comment]
Current dependencies on/for this PR:
* main
  * **PR #73** <a href="https://app.graphite.dev/github/pr/aptos-labs/aptos-ts-sdk/73" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/aptos-labs/aptos-ts-sdk/73?utm_source=stack-comment).

File: src/api/aptos.ts [hunk 1]
```diff
@@ -106,6 +111,7 @@ applyMixin(Aptos, Coin, "coin");
 applyMixin(Aptos, Collection, "collection");
 applyMixin(Aptos, Event, "event");
 applyMixin(Aptos, Faucet, "faucet");
+applyMixin(Aptos, FungibleAsset, "fungible_asset");
```
Comment (gregnazario) [inline src/api/aptos.ts:114]: Let's avoid snake case and have everything use camelcase.  I'm going to change the rest but it ends up with a lot of conflicts

File: src/api/fungible_asset.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  }
+
+  /**
+   * Queries the fungible asset metadata
+   *
+   * This query returns the fungible asset metadata for all fungible assets.
+   * It can be filtered by creator address and asset type.
+   *
+   * @param options.pagination Where offset and limit can be specified
+   * @param options.whereCondition Where condition to filter the results
```
Comment (gregnazario) [inline src/api/fungible_asset.ts:25]: These should be args.options.pagination etc

File: src/api/fungible_asset.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+   * @returns getFungibleAssetMetadata A list of fungible asset metadata
+   */
+  async getFungibleAssetMetadata(args: {
+    options?: {
+      pagination?: PaginationArgs;
+      whereCondition?: {
```
Comment (gregnazario) [inline src/api/fungible_asset.ts:31]: We need to look at what we name this, elsewhere it's simply where

File: src/api/fungible_asset.ts [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+ */
+export class FungibleAsset {
+  readonly config: AptosConfig;
+
+  constructor(config: AptosConfig) {
+    this.config = config;
+  }
+
+  /**
+   * Queries the fungible asset metadata
```
Comment (gregnazario) [inline src/api/fungible_asset.ts:19]: Queries the current fungible asset metadata.  Makes it clearer that the  indexer only returns (semi current state)

File: src/api/fungible_asset.ts [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+   * @param options.pagination Where offset and limit can be specified
+   * @param options.whereCondition Where condition to filter the results
+   * @returns getFungibleAssetMetadata A list of fungible asset metadata
+   */
+  async getFungibleAssetMetadata(args: {
+    options?: {
+      pagination?: PaginationArgs;
+      whereCondition?: {
+        creatorAddress?: HexInput;
+        assetType?: string;
```
Comment (gregnazario) [inline src/api/fungible_asset.ts:33]: Can we use the structured type (0x{}::{}::{}) I think it's somewhere else in the code

File: src/api/aptos.ts (hunk 1 above)
Comment (0xjinn) [reply src/api/aptos.ts:114]: hmm wait, thats the class name, I think it make sense for snake case? Similar to `TransactionSubmission` on line 109

File: src/api/fungible_asset.ts (hunk 3 above)
Comment (0xjinn) [reply src/api/fungible_asset.ts]: Updated to use bool exp

File: tests/e2e/api/fungible_asset.test.ts [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+// SPDX-License-Identifier: Apache-2.0
+
+import { Aptos, AptosConfig, Network } from "../../../src";
+import { FungibleAssetMetadataBoolExp } from "../../../src/types/generated/types";
+
+describe("FungibleAsset", () => {
+  test("it should fetch fungible asset metadata", async () => {
+    const config = new AptosConfig({ network: Network.LOCAL });
+    const aptos = new Aptos(config);
+    const APT_COIN_TYPE = "0x1::aptos_coin::AptosCoin";
```
Comment (heliuchuan) [inline tests/e2e/api/fungible_asset.test.ts:11]: i assume you'll handle this somewhere in pr stack

**Review by heliuchuan — APPROVED** [review]

**Pull request merged**